src/Castle.Windsor/MicroKernel/Arguments.cs

```diff
@@ -26,12 +26,9 @@ namespace Castle.MicroKernel
 	/// </summary>
 	public class Arguments : IDictionary
 	{
-		protected IDictionary arguments;
+		private bool isReadOnly = false;
 
-		public Arguments(object namedArgumentsAsAnonymousType, params IArgumentsComparer[] customComparers)
-			: this(new ReflectionBasedDictionaryAdapter(namedArgumentsAsAnonymousType), customComparers)
-		{
-		}
+		protected IDictionary arguments;
```

Comment:
The `arguments` field used to be protected but I see no reason to leave it protected, I think we should make it private.

Can we also now mark the field readonly since the internal collection is only ever created in the constructor now that we don't maintain the dictionary passed in.